src/sync.cpp

```diff
@@ -113,7 +103,7 @@ static void potential_deadlock_detected(const std::pair<void*, void*>& mismatch,
 
 static void push_lock(void* c, const CLockLocation& locklocation)
 {
-    LockData& lockdata = GetLockData();
+    LockData& lockdata{*g_lockdata.Get()};
```

Comment:
I prefer following code rather than calling Get() directly.
```suggestion
    LockData& lockdata = *g_lockdata;
```
or
```suggestion
    LockData& lockdata{*g_lockdata};
```